web/modules/custom/joinup_news/src/Guard/FulfillmentGuard.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+namespace Drupal\joinup_news\Guard;
+
+use Drupal\node\NodeInterface;
+use Drupal\og\Og;
+use Drupal\state_machine\Guard\GuardInterface;
+use Drupal\state_machine\Plugin\Workflow\WorkflowInterface;
+use Drupal\state_machine\Plugin\Workflow\WorkflowTransition;
+use Drupal\Core\Entity\EntityInterface;
+
+/**
+ * Guard class for the transitions of the news entity.
+ *
+ * @package Drupal\joinup_news\Guard
+ */
+class FulfillmentGuard implements GuardInterface {
+  /**
+   * Constant representing the pre-moderating parent.
+   */
+  const MODERATED = 1;
+
+  /**
+   * Constant representing the post-moderating parent.
+   */
+  const NON_MODERATED = 0;
+
+  /**
+   * Virtual state.
+   */
+  const NON_STATE = '__new__';
+
+  /**
+   * {@inheritdoc}
+   *
+   * We need to override default transitions allowed because this is also
+   * dependant on the parent's moderation, system roles and organic groups
+   * user roles. In the following method, the allowed transitions per
+   * moderation are checked and then if the transition is allowed, the user
+   * roles by the system and the organic groups are checked.
+   */
+  public function allowed(WorkflowTransition $transition, WorkflowInterface $workflow, EntityInterface $entity) {
+    $to_state = $transition->getToState()->getId();
+    // Disable virtual state.
+    if ($to_state == self::NON_STATE) {
+      return FALSE;
+    }
+
+    $from_state = $this->getState($entity);
+    $parent = joinup_news_get_parent($entity);
+
+    $is_moderated = self::MODERATED;
+    if ($parent) {
+      $is_moderated = ($parent->bundle() == 'collection') ?
+        $parent->field_ar_moderation->first()->value :
+        $parent->field_is_moderation->first()->value;
+    }
+    $allowed_conditions = \Drupal::config('joinup_news.settings')->get('transitions');
+
+    // Some transitions are not allowed per parent's moderation.
+    // Check for the transitions allowed.
+    // @see: joinup_news.settings.yml
+    if (!isset($allowed_conditions[$is_moderated][$to_state][$from_state])) {
+      return FALSE;
+    }
+
+    // This Guard class's method called whenever the transitions are checked
+    // even outside the entity CRUD forms. Cases like this is e.g. when trying
+    // to edit the settings of the field.
+    // In these cases, there is no parent entity so we need to check for it.
+    if ($parent) {
```

Comment:
I think it would be more readable to do a `if (!$parent) { return FALSE; }` as there would be no further processing anyway.